# Students Dataset
**Here we are going to Analyse the data using pandas and**

- Importing pandas as pd(Alias) to perform tasks)
- importing dataset into 'df'

In [89]:
import pandas as pd

df=pd.read_csv(r"C:\Users\hp\Downloads\student_dataset.csv")
df.head()

StudentID    Name  Gender           Course  Age  Marks Grade       City
0       2001    Isha  Female     Data Science   27     82     F  Bangalore
1       2002   Aarav  Female     Data Science   27     42     A  Hyderabad
2       2003  Vivaan    Male  Web Development   24     98     B    Kolkata
3       2004   Sanya  Female  Web Development   23     52     A  Hyderabad
4       2005   Anaya  Female  Web Development   28     51     A  Hyderabad

**Performing shape to find out number of rows and columns**

In [93]:
df.shape

(150, 8)

- As per shape, there are 150 rows and 8 columns present in the dataset

**Performing Numerical calculations using describe method**

In [3]:
df.describe()

StudentID         Age       Marks
count   150.000000  150.000000  150.000000
mean   2075.500000   24.346667   67.033333
std      43.445368    3.701962   17.598206
min    2001.000000   18.000000   40.000000
25%    2038.250000   21.250000   52.000000
50%    2075.500000   24.000000   62.000000
75%    2112.750000   27.000000   82.750000
max    2150.000000   30.000000  100.000000

**As per the data**

- Student id is between 2001-2150
- Student's min age is 18 and max age is 30
- min marks obtained by student is 40 and max marks are 100

**Taking information about the dataset using info method**

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 8 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   StudentID  150 non-null    int64 
 1   Name       150 non-null    object
 2   Gender     150 non-null    object
 3   Course     150 non-null    object
 4   Age        150 non-null    int64 
 5   Marks      150 non-null    int64 
 6   Grade      150 non-null    object
 7   City       150 non-null    object
dtypes: int64(3), object(5)
memory usage: 9.5+ KB


- Here are 8 columns having int and object datatypes
- there are 150 values in each columns

**Checking if there is any null value or missing value in each column**

In [21]:
df.isnull().sum()

StudentID    0
Name         0
Gender       0
Course       0
Age          0
Marks        0
Grade        0
City         0
dtype: int64

- there aren't any null or missing value in any column

**Checking if any duplicated value present in student id column**

In [23]:
df['StudentID'].duplicated().sum()

0

- There isn't duplicate value in student id column

**Count of number of same gender in the column 'Gender'**

In [35]:
df['Gender'].value_counts()

Gender
Male      77
Female    73
Name: count, dtype: int64

- As per data, there are 77 Male and 73 Female students

**Counting how many students are having same age**

In [49]:
df['Age'].value_counts().sort_index()

Age
18    12
19    10
20     7
21     9
22     6
23    17
24    16
25    14
26    10
27    12
28    12
29    10
30    15
Name: count, dtype: int64

- So according to the data, we have 13 different ages
- starts from 18 and ends at 30
- by maually grouping we can say that,
- 77 student's are between 18-24 age group
- 73 student's are between 25-30 age group

**Counting values for each unique grade**

In [47]:
df['Grade'].value_counts().sort_index()

Grade
A    36
B    24
C    33
D    32
F    25
Name: count, dtype: int64

- We have 5 grades according to marks
- 'A', 'B', 'C', 'D', 'F' are the grades
- 36, 24, 33, 32, 25 are the count of students respectively for the grades
  

**Counting the students belongs to their respective city**

In [51]:
df['City'].value_counts().sort_index()

City
Bangalore    48
Delhi        17
Hyderabad    28
Kolkata      28
Mumbai       29
Name: count, dtype: int64

- according to the data,
- most of the students belongs to Bangalore city(48)
- followed by mumbai(29), hyderabad and kolkata 28 for each and delhi with 17

**Finding no. of students who got 100 marks**

In [75]:
df[df['Marks'] == 100]

StudentID   Name  Gender           Course  Age  Marks Grade       City  \
17       2018  Aarav  Female     Data Science   27    100     C    Kolkata   
36       2037   Diya  Female  Web Development   29    100     C  Bangalore   

   AgeGroup  
17    26-30  
36    26-30

- There are only 2 students who got 100 marks

**Finding no. of students who got marks above 90**

In [85]:
(df['Marks'] >= 90).value_counts()

Marks
False    125
True      25
Name: count, dtype: int64

- There are 25 student's who got marks above 90(including the student's who got 100)
- soo there are 23 students between 90-99 range

In [132]:
(df['Marks'] >= 80).value_counts()

Marks
False    108
True      42
Name: count, dtype: int64

- if we remove 25 students who got marks above 90, there are 17 students who got marks between 80-89

In [134]:
(df['Marks'] >= 70).value_counts()

Marks
False    89
True     61
Name: count, dtype: int64

- similarly there are 19 students who got marks between 70-79

In [136]:
(df['Marks'] >= 60).value_counts()

Marks
True     86
False    64
Name: count, dtype: int64

- 25 students got marks between 60-69

In [138]:
(df['Marks'] >= 50).value_counts()

Marks
True     125
False     25
Name: count, dtype: int64

- 39 students got marks between 50-59

In [145]:
(df['Marks'] >= 40).value_counts()

Marks
True    150
Name: count, dtype: int64

- Similarly 25 students got marks between 40-49